sources/packages/backend/apps/api/src/route-controllers/application/application.aest.controller.ts

```diff
@@ -231,4 +233,28 @@ export class ApplicationAESTController extends BaseController {
       applicationId,
     );
   }
+
+  /**
+   * Get history of application versions for an application
+   * where the current application is excluded.
+   * @param applicationId application Id.
+   * @returns history of application versions.
+   */
+  @ApiNotFoundResponse({
+    description: `Application not found or not on ${ApplicationStatus.Completed} status.`,
+  })
+  @Get(":applicationId/versions")
+  async getApplicationVersionHistory(
+    @Param("applicationId", ParseIntPipe) applicationId: number,
+  ): Promise<ApplicationVersionAPIOutDTO[]> {
+    const applications =
+      await this.applicationService.applicationVersionHistory(applicationId);
+    if (!application) {
```

Comment:
To validate if the application exist, the service has a method `doesApplicationExist`. Please use the same.

```ts
const application = await this.applicationService.doesApplicationExist({
      applicationId,
    });
    if (!application) {
      throw new NotFoundException("Application not found.");
    }
``